frame/executive/src/lib.rs

```diff
@@ -101,6 +98,43 @@ pub type CheckedOf<E, C> = <E as Checkable<C>>::Checked;
 pub type CallOf<E, C> = <CheckedOf<E, C> as Applyable>::Call;
 pub type OriginOf<E, C> = <CallOf<E, C> as Dispatchable>::Origin;
 
+
+/// The last time the runtime upgrade happened.
+///
+/// Stores the the `impl_version` and the `spec_version` of this runtime.
```

Comment:
it should store `spec_name` and `spec_version`; `impl_` stuff is strictly unneeded according to our version semantics.